tools/Google.Cloud.Tools.Analyzers/InternalOptionalParametersRequiredAnalyzer.cs

```diff
@@ -106,5 +85,38 @@ private static void AnalyzeInvocation(OperationAnalysisContext context)
                 }
             }
         }
+
+        internal static List<ISymbol> GetVariablesInScope(SyntaxNode syntaxNode, SemanticModel semanticModel) =>
```

Comment:
Well that just means that if there are multiple omitted arguments per call, we would need to re-iterate over all symbols in scope to find just locals or parameters, but for the common case or 0 or 1 omitted arguments, we would perform the iteration 0 or 1 times, respectively, so yes overall this is much better.